OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectList.cs

```diff
@@ -200,15 +231,86 @@ sealed class DreamMetaObjectList : IDreamMetaObject {
+                            DreamValue associativeValue = list.GetValue(value);
+
+                            newList.AddValue(value);
+                            if (associativeValue != DreamValue.Null) newList.SetValue(value, associativeValue);
+                        }
+                    }
+                } else {
+                    int len = list.GetLength();
+
+                    for (int i = 1; i <= len; i++) {
+                        DreamValue value = list.GetValue(new DreamValue(i));
+
+                        if (value == b) {
+                            DreamValue associativeValue = list.GetValue(value);
+
+                            newList.AddValue(value);
+                            if (associativeValue != DreamValue.Null) newList.SetValue(value, associativeValue);
+                        }
+                    }
+                }
+
+                state.Push(new DreamValue(newList));
+            }
+            else
+            {
+                state.Push(new DreamValue(0));
+            }
+            return null;
         }
 
-        public DreamValue OperatorIndex(DreamObject dreamObject, DreamValue index) {
-            return ((DreamList)dreamObject).GetValue(index);
+        public ProcStatus? OperatorBitXor(DreamValue a, DreamValue b, DMProcState state) {
+            DreamList list = a.MustGetValueAsDreamList();
+            DreamList newList = _objectTree.CreateList();
+            List<DreamValue> values;
+
+            if (b.TryGetValueAsDreamList(out DreamList secondList)) {
+                values = secondList.GetValues();
+            } else {
+                values = new List<DreamValue>() { b };
+            }
+
+            foreach (DreamValue value in values) {
+                bool inFirstList = list.ContainsValue(value);
+                bool inSecondList = secondList.ContainsValue(value);
+
+                if (inFirstList ^ inSecondList) {
+                    newList.AddValue(value);
+
+                    DreamValue associatedValue = inFirstList ? list.GetValue(value) : secondList.GetValue(value);
+                    if (associatedValue != DreamValue.Null) newList.SetValue(value, associatedValue);
+                }
+            }
+
+            state.Push(new DreamValue(newList));
+            return null;
+        }
+        public ProcStatus? OperatorIndex(DreamValue a, DreamValue index, DMProcState state) {
+            DreamList dreamList = a.MustGetValueAsDreamList();
+            state.Push(dreamList.GetValue(index));
+            return null;
         }
 
-        public void OperatorIndexAssign(DreamObject dreamObject, DreamValue index, DreamValue value) {
-            ((DreamList)dreamObject).SetValue(index, value);
+        public ProcStatus? OperatorIndexAssign(DreamValue a, DreamValue index, DreamValue value, DMProcState state) {
+            DreamList dreamList = a.MustGetValueAsDreamList();
+            dreamList.SetValue(index, value);
+            state.Push(value);
```

Comment:
The `Assign` opcode already pushes the value onto the stack, this leads to two values being pushed when there should only be one.
```suggestion
```